### PR TITLE
(#93) make module implements io.WriterTo

### PR DESCRIPTION
**Pull request by dannypsnl**

### CODE REVIEW

File: ir/module.go [hunk 1]
```diff
@@ -187,6 +188,160 @@ func (m *Module) String() string {
...
+		// 'target' 'datalayout' '=' DataLayout=StringLit
+		writeWrapping.Fprintf(output, "target datalayout = %s\n", quote(m.DataLayout))
+	}
+	// Target triple.
+	if len(m.TargetTriple) > 0 {
+		// 'target' 'triple' '=' TargetTriple=StringLit
+		writeWrapping.Fprintf(output, "target triple = %s\n", quote(m.TargetTriple))
+	}
+	// Module-level inline assembly.
+	if len(m.ModuleAsms) > 0 {
```
Comment (mewmew) [inline ir/module.go]: Just noticed that the `&& buf.Len() > 0`  check was removed from these lines. The equivalent line would be `&& writeWrapping.size > 0`.

File: ir/module.go [hunk 2]
```diff
@@ -187,6 +188,160 @@ func (m *Module) String() string {
...
 }
 
+type ioWriteWrapping struct {
+	size int
+	err  error
+}
+
+func (i *ioWriteWrapping) Fprintln(w io.Writer, a ...interface{}) {
+	if i.err != nil {
+		n, err := fmt.Fprintln(w, a)
```
Comment (mewmew) [inline ir/module.go]: Also noticed now with `go vet` that this line should include `a...`

```
./module.go:205:13: missing ... in args forwarded to print-like function
```

```diff
- 		n, err := fmt.Fprintln(w, a) 
+ 		n, err := fmt.Fprintln(w, a...) 
```

Comment (dannypsnl) [reply ir/module.go]: forgot the test, I'm fixing it

File: ir/module.go [hunk 3]
```diff
@@ -187,6 +188,160 @@ func (m *Module) String() string {
 	return buf.String()
 }
 
+type ioWriteWrapping struct {
+	size int
+	err  error
+}
+
+func (i *ioWriteWrapping) Fprintln(w io.Writer, a ...interface{}) {
+	if i.err != nil {
```
Comment (mewmew) [inline ir/module.go]: Also, taking a closer look, I think that this check should be inverted, we only want to invoke further `fmt.Fprintln` calls if there has *not* yet been an error.

So either:

```diff
- 	if i.err != nil { 
+ 	if i.err == nil { 
```

Or simply an early return:

```go
if i.err != nil { 
   return
}
```

Comment (dannypsnl) [reply ir/module.go]: I'm not pretty sure with this, for me, it means `if no error`

Comment (mewmew) [reply ir/module.go]: Without inverting the check, the `fmt.Fprintln` call would never run, as `i.err` is initialized to `nil`. This the code inside the `if err != nil { }` block would never run.

Comment (dannypsnl) [reply ir/module.go]: Oh, I got it, a typo bug.

**Pull request merged**